src/app/modules/main/communities/private_interfaces/module_access_interface.nim

```diff
@@ -0,0 +1,82 @@
+  raise newException(ValueError, "No implementation available") 
+
+method editCommunityCategory*(self: AccessInterface, communityId: string, categoryId: string, name: string, channels: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method deleteCommunityCategory*(self: AccessInterface, communityId: string, categoryId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method reorderCommunityCategories*(self: AccessInterface, communityId: string, categoryId: string, position: int) {.base} =
+  raise newException(ValueError, "No implementation available") 
+
+method reorderCommunityChannel*(self: AccessInterface, communityId: string, categoryId: string, chatId: string, position: int) {.base} =
+  raise newException(ValueError, "No implementation available") 
+
+method leaveCommunity*(self: AccessInterface, communityId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method inviteUsersToCommunityById*(self: AccessInterface, communityId: string, pubKeysJSON: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method removeUserFromCommunity*(self: AccessInterface, communityId: string, pubKey: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method banUserFromCommunity*(self: AccessInterface, pubKey: string, communityId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method requestToJoinCommunity*(self: AccessInterface, communityId: string, ensName: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+  
+method acceptRequestToJoinCommunity*(self: AccessInterface, communityId: string, requestId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method declineRequestToJoinCommunity*(self: AccessInterface, communityId: string, requestId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method requestCommunityInfo*(self: AccessInterface, communityId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method deleteCommunityChat*(self: AccessInterface, communityId: string, channelId: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method setCommunityMuted*(self: AccessInterface, communityId: string, muted: bool) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method importCommunity*(self: AccessInterface, communityKey: string) {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+method exportCommunity*(self: AccessInterface, communityId: string): string {.base.} =
+  raise newException(ValueError, "No implementation available") 
+
+type
+  ## Abstract class (concept) which must be implemented by object/s used in this 
+  ## module.
+  DelegateInterface* = concept c
```

Comment:
@jrainville since your class (module) is not a template class any more you don't need this concept thing here. Remove `DelegateInterface` concept type completely from here.